test/signal_test.dart

```diff
@@ -0,0 +1,385 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:cli_script/cli_script.dart';
+import 'package:test/test.dart';
+
+import 'util.dart';
+
+void main() {
+  group('on Script', () {
+    test('interrupts with default signal', () async {
+      var script =
+          mainScript('await Future<void>.delayed(Duration(seconds: 100));');
+
+      expect(await script.kill(), true);
+
+      expect(script.done, throwsScriptException(-15));
```

Comment:
It's probably good to verify that this didn't take 100 seconds to trigger as well.

Reply:
Rather than giving the test a short timeout, which runs the risk of making it flaky, you could start a stopwatch before `script.kill()` and then verify that it's below some threshold after `script.done`. Alternatively, you could have the `mainScript()` wait _forever_, rather than for a set amount of time, for example by a `while (true)` loop with an await inside it.